packages/client/src/types/config.types.ts

```diff
@@ -1,3 +1,13 @@
-import type { AxiosRequestConfig } from 'axios';
+import type { AxiosRequestConfig, AxiosRequestHeaders } from 'axios';
 
-export type Config = AxiosRequestConfig & Record<string, unknown>;
+type MockServerProps = {
+  'X-MOCK-SERVER-IGNORE'?: boolean;
+};
+
+type CustomHeaders = {
+  headers?: AxiosRequestHeaders & MockServerProps;
+};
+
+export type Config = AxiosRequestConfig &
```

Comment:
You will need to Omit the `headers` property from `AxiosRequestConfig` since they will clash here with the type defined on `CustomHeaders` type like this:

```suggestion
export type Config = Omit<AxiosRequestConfig, 'headers'> &
```

Reply:
A message like that: `Types of property 'headers' are incompatible`.
Now, I'm using the right types and it was fixed.